mne_features/univariate.py

```diff
@@ -976,19 +978,22 @@ def _zero_crossings(data):
     zc = np.zeros((n_channels,), dtype=data.dtype)
     for j in range(n_channels):
         for i in range(n_times - 1):
-            if data[j, i] == 0 or data[j, i] * data[j, i + 1] < 0:
+            if abs(data[j, i]) < threshold or data[j, i] * data[j, i + 1] < 0:
```

Comment:
this does not seem right. If you gets positive but very close to 0. but never gets negative what happens? For me the behavior should be that you consider a zero crossing if the signal goes from +threshold to -threshold eventually over multiple samples.

Reply:
In the [example](https://github.com/mne-tools/mne-features/issues/45), the last two values of `y` are:  `-3.42020143e-01, -9.79717439e-16`. The last value is actually `sin(8 * pi)`, which is mathematically 0. Yet, the value `-9.79717439e-16` was not considered to be a zero crossing since its sign is < 0. The `abs(data[j, i]) < threshold` was intended to clip all values in ]-threshold, threshold[ to 0. As a result, yes, if the signal gets very close to 0, `abs(data[j, i]) < threshold` might always be True. To me, the behavior of the function depends on the threshold and, to avoid any problem, the threshold should be chosen equal (or close to) numerical/machine precision.

Should we raise a warning if a threshold different than `np.finfo(np.float64).eps` is used?

Reply:
no need to add a warning. If you change this value it's your responsability.

now what worries me more is that if you have consecutive values below threshold then you seem to count multiple crossings. For me you have a crossing if you goes from +thresh to -thresh over consecutive time instants

Reply:
I think I see what you mean : if `data = np.array([1., 0.5, 0., 2.4e-17., -1., -9.8e-16])`, you would expect the function to count 2 zero-crossings (`data[2]` and `data[-1]`) instead of 3 (`data[2]`, `data[3]` and `data[-1]`) ?